components/send-batch-status-modal.tsx

```diff
@@ -89,14 +89,24 @@ export const SendBatchStatusModal: React.FC = () => {
           </Text>
         </Group>
         <Progress value={progress} animated={status === 'sending'}></Progress>
-        <Button
-          disabled={status === 'sending'}
-          onClick={() => {
-            useSendBatchState.getState().clear()
-          }}
-        >
-          Dismiss
-        </Button>
+        {(status === 'cancelled' || status === 'finished') && (
+          <Button
+            onClick={() => {
+              useSendBatchState.getState().clear()
+            }}
+          >
+            Dismiss
+          </Button>
+        )}
+        {status === 'sending' && (
+          <Button
+            onClick={() => {
+              useSendBatchState.getState().cancel()
+            }}
+          >
+            Cancel
+          </Button>
+        )}
```

Comment:
Does it make more sense to use `status === 'sending' ? <Button with cancel /> : <Button with dismiss />`?